capstone-app/src/app/modules/maps/map/map.component.ts

```diff
@@ -110,59 +127,48 @@ export class MapComponent implements OnInit, OnChanges {
     this.activeMarkerOutput.emit(this.activeMark);
   }
 
-  getCurrentLocation () {
-    let filterResult: string;
-    let prevResult: string = 'blank';
-    navigator.geolocation.getCurrentPosition(position => {
-      this.position = {
-        lat: position.coords.latitude,
-        lng: position.coords.longitude,
-      };
-      this.location = new google.maps.LatLng(this.position);
-      this.geocoder.geocode({'location': this.position}, (results, status) => {
-        if (status === 'OK') {
-          for (let result of results) {
-            filterResult = result.formatted_address;
-            if (filterResult.match(/[1-1000]/)) {
-              continue;
-            }
-            if (filterResult.includes('County')){
-              continue;
-            }
-            if (!filterResult.includes(',')) {
-              continue;
-            }
-            if (filterResult.split(',').length > 3) {
-              continue;
-            }
-            if (filterResult.split(',').length > prevResult.split(',').length) {
-              this.cityLocation = filterResult;
-            }
-            else {
-              this.cityLocation = prevResult;
-            }
-            prevResult = filterResult;
-          }
-        }
-        this.cityOutput.emit(this.cityLocation);
-      });
+  private locationCallbackFail() {
+    this.cityLocation = "Los Angeles, CA, USA"
+      this.location = new google.maps.LatLng(34.0522, -118.2437);
       this.placesRequestFunc(this.location);
-    });
-    
+      // change to this.placeCitySharer.setPlaces(this.cityLocation);
+      this.placeCitySharer.setCityName(this.cityLocation);
+    }
+
+  locationCallbackSuccess(position: Position) {
+    this.position = {
+          lat: position.coords.latitude,
+          lng: position.coords.longitude,
+        };
+        this.location = new google.maps.LatLng(this.position);
+        this.geocoder.geocode({'location': this.position}, (results, status) => {
+          if (status === 'OK') {
+            const resultArr = results.filter(result => result.types.includes('locality'));
+            this.cityLocation = resultArr[0].formatted_address;
+          }
+          // change to this.placeCitySharer.setPlaces(this.cityLocation);
+          console.log(this.cityLocation);
+          this.placeCitySharer.setCityName(this.cityLocation);
+        });
+        this.placesRequestFunc(this.location);
+  }
+
+  getCurrentLocation () {
+    navigator.geolocation.getCurrentPosition(position =>  this.locationCallbackSuccess(position), this.locationCallbackFail);
```

Comment:
Nit: There are two space between the arrow and this.locationCallbackSuccess. Remove one.